org.lflang/src/org/lflang/generator/GeneratorBase.xtend

```diff
@@ -440,6 +443,41 @@ abstract class GeneratorBase extends AbstractLFValidator {
         // use the result.
     }
 
+    /**
+     * Check if a clean was requested from the standalone compiler and perform the clean step.
+     */
+    protected def void cleanIfNeeded(IGeneratorContext context) {
```

Comment:
Seem like a better place to put this would be `FileConfig`. Note that `context` is also accessible in that class.

Reply:
The actual cleaning is target dependent, so it wouldn't really make sense to factor this out. I think what could be factored out is the check whether cleaning is needed or not. I am not sure if `FileConfig` is really the right place for it, but I could add a flag `needsClean` or something similar there. Probably the `deleteDirectory` should also be moved to `FileConfig`.

Reply:
Yes, my suggestion would be to move all of these new methods to `FileConfig`. Note that we can have target-specific extensions of `FileConfig`; the TypeScript generator so far is the only one that needed it, but we can easily create one for Cpp as well.

Reply:
Can you point me to an example of how this extension mechanism works? I did not find it in `FileConfig.java` nor in `TypeScriptGenerator.xtend`.

Reply:
[`FileScriptFileConfig.java`](https://github.com/icyphy/lingua-franca/blob/master/org.lflang/src/org/lflang/generator/TypeScriptFileConfig.java) is a specialized version of `FileConfig`, tailored to TypeScript. `TypeScriptGenerator` [overrides](https://github.com/icyphy/lingua-franca/blob/9bbb747d6e32db816c3420d3895a0842bdfb6d3b/org.lflang/src/org/lflang/generator/TypeScriptGenerator.xtend#L1370) `setFileConfig` in order to assign an instance of `TypeScriptFileConfig` to the `fileConfig` class variable specified in `GeneratorBase`.

Reply:
I moved the code over to `FileConfig`